### ISSUE TITLE
False positive CS9217 on ref assignment from conditional expression after await

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn#83727]
**Version Used**: 18.5.3, [.NET Lab](https://lab.razor.fyi/#41rEyMUVUJSfXpSYq5dcLDSZsbQ4My9dIbiyuCQ1Vy8koyg1MSUzL10vJLE4u9iaiyuxuDIvWQHEU_DVyMwrUajQ5KrmUlBQUChKTVMACRQp2ILZFdZg4cTyxMwSsAa9yMzUnBQNTYg4SGmlgq2CIYQL0lVSVJqqYA_WXKlgBaGtuWq5vJiSi6M4OY70L7l1NF2AJYGxghEA)

```cs
using System.Threading.Tasks;

async Task M(int x)
{
    ref int r = ref x;
    await Task.Yield();
    int y = 1;
    // error CS9217: A 'ref' local cannot be preserved across 'await' or 'yield' boundary.
    r = true ? ref y : ref y;
}
```

Yet the ref is not being preserved across the boundary. Reassignment is allowed. 

For comparison, this code correctly compiles:
```cs
using System.Threading.Tasks;

async Task M(int x)
{
    ref int r = ref x;
    await Task.Yield();
    int y = 1;
    r = ref y;
}
```

**Comment by jnm2** [comment]
User error, the snippet actually does dereference `r` to assign it. The intended code was:

```cs
async Task M(int x)
{
    ref int r = ref x;
    await Task.Yield();
    int y = 1;
    r = ref (true ? ref y : ref y);
}
```
